src/views/view.c

```diff
@@ -862,7 +865,8 @@ const GList *dt_view_get_images_to_act_on(const gboolean only_visible, const gbo
           // care of eventual duplicates
           l = g_list_prepend(l, GINT_TO_POINTER(sqlite3_column_int(stmt, 0)));
         }
-        // put the list in right order as we have prepend for performance reasons
+        // put back the list in right order as we have prepend items for perf reasons
+        l = g_list_reverse(l);
```

Comment:
In fist case keeping DESC we could avoid the g_list_reverse(). What about using a boolean and reverse the list only in the second case?